src/sinol_make/helpers/compile.py

```diff
@@ -25,6 +25,8 @@ def compile(program, output, compilers: Compilers = None, compile_log = None, we
     """
     if extra_compilation_args is None:
         extra_compilation_args = []
+    if isinstance(extra_compilation_args, str):
+        extra_compilation_args = [extra_compilation_args]
```

Comment:
Usually, after such modifications (which serve the purpose of unifying the types to a single one), we can make the code more readable by exactly saying what the type should be, e.g. by making an `assert(isinstance(extra_compilation_args, Array[string]))` (handwritten, might not work)

Reply:
What do you think about this assert `assert isinstance(extra_compilation_args, list) and all(isinstance(arg, str) for arg in extra_compilation_args)`? It's quite long, but there no other way to check if the type is list of strings

Reply:
It's good enough, I guess.